sdk/core/Azure.Core/tests/DynamicData/MutableJsonElementTests.cs

```diff
@@ -548,6 +549,36 @@ public void CanGetDateTimeOffset()
             Assert.Throws<InvalidOperationException>(() => mdoc.RootElement.GetProperty("foo").GetDateTimeOffset());
         }
 
+        [Test]
+        public void StaticMethodsAreSameAsExpected()
```

Comment:
Can you help me understand what these tests are doing?  Is it just to ensure that if we add APIs to MJD/MJE that we annotate them correctly for AOT?  i.e. these will fail when we add something and the message will notify the person making the change what to do?

Reply:
This could be moot after your changes to those classes - but before this the reason was that:
* MutableJsonElement is a struct - currently RequiresUnreferencedCode can't be applied to struct (only class) - this is a limitation the tooling (it's just pretty difficult to detect all possible calls to struct .ctor since lot of them are implicit).
* We needed the same behavior as putting RequiresUnrereferencedCode onto the type - meaning any access to .ctor or static members is considered problematic.
* So to emulate that we added RequiresUnrereferencedCode onto the constructor and verified that the type doesn't have problematic static methods.
* We used warning suppressions under the assumption that all entry points into the type (.ctor or static) are covered.
* That assumption is "soft" - not-enforcable by tooling. So I suggested to add a unit test which validates it. The goal of the test is to fail is somebody modifies the class to potentially violate the above assumption. It's mostly there to make them think about the trimming problem in that class. We use similar unit tests in other projects to do similar things and it already proved useful.